rest/config.go

```diff
@@ -61,43 +62,46 @@ const (
+	CORS                         *CORSConfig              `json:",omitempty"`                                 // Configuration for allowing CORS
+	DeprecatedLog                []string                 `json:"log,omitempty"`                              // Log keywords to enable
+	DeprecatedLogFilePath        *string                  `json:"logFilePath,omitempty"`                      // Path to log file, if missing write to stderr
+	Logging                      *base.LoggingConfig      `json:",omitempty"`                                 // Configuration for logging with optional log file rotation
+	Pretty                       bool                     `json:",omitempty"`                                 // Pretty-print JSON responses?
+	DeploymentID                 *string                  `json:",omitempty"`                                 // Optional customer/deployment ID for stats reporting
+	StatsReportInterval          *float64                 `json:",omitempty"`                                 // Optional stats report interval (0 to disable)
+	CouchbaseKeepaliveInterval   *int                     `json:",omitempty"`                                 // TCP keep-alive interval between SG and Couchbase server
+	SlowQueryWarningThreshold    *int                     `json:",omitempty"`                                 // Log warnings if N1QL queries take this many ms
+	MaxIncomingConnections       *int                     `json:",omitempty"`                                 // Max # of incoming HTTP connections to accept
+	MaxFileDescriptors           *uint64                  `json:",omitempty"`                                 // Max # of open file descriptors (RLIMIT_NOFILE)
+	CompressResponses            *bool                    `json:",omitempty"`                                 // If false, disables compression of HTTP responses
+	Databases                    DbConfigMap              `json:",omitempty"`                                 // Pre-configured databases, mapped by name
+	Replications                 []*ReplicateV1Config     `json:",omitempty"`                                 // sg-replicate replication definitions
+	MaxHeartbeat                 uint64                   `json:",omitempty"`                                 // Max heartbeat value for _changes request (seconds)
+	ClusterConfig                *ClusterConfig           `json:"cluster_config,omitempty"`                   // Bucket and other config related to CBGT
+	Unsupported                  *UnsupportedServerConfig `json:"unsupported,omitempty"`                      // Config for unsupported features
+	ReplicatorCompression        *int                     `json:"replicator_compression,omitempty"`           // BLIP data compression level (0-9)
+	ActiveReplicatorPingInterval *int                     `json:"replicator_ping_interval_seconds,omitempty"` // Websocket Ping interval (for active replicators). Defaults to 5 minutes. Set to 0 to disable pings.
```

Comment:
I'm on the fence about whether this should be a server config setting, since sg-replicate2 is otherwise completely database scoped.

Reply:
Fair enough, given it doesn't affect passive replicator connections. Moved to be alongside `sgreplicate_enabled`.